firedrake/mesh.py

```diff
@@ -2366,6 +2381,57 @@ def _pic_swarm_in_plex(plex, coords, fields=[]):
     return swarm
 
 
+def _parent_mesh_embedding(vertex_coords, parent_mesh):
+    """Find the parent cells and local coords for vertices on this rank."""
```

Comment:
You should specifically note that this never puts any vertices in the parent mesh ghost cells I think